app/jobs/csv_export_job.rb

```diff
@@ -0,0 +1,65 @@
+require 'csv'
+
+class CsvExportJob < ActiveJob::Base
+  queue_as :csv_jobs
+ 
+  def perform(csv_export)
+    ActiveRecord::Base.connection_pool.with_connection do
+      create_export_folder(csv_export)
+      export_task(csv_export)
+      cleanup_downloaded
+    end
+  end
+  
+  private
+  
+  def cleanup_downloaded
+    # TODO delete files with no record associations
+    CsvExport.old.each do |csv_export|
+      csv_export.destroy
+    end
+  end
+  
+  def export_task(csv_export)
+    csv_export.status! :started
+    deploy_csv_export(csv_export)
+    notify_of_creation(csv_export)
+  rescue NoMethodError, IOError, ActiveRecord::ActiveRecordError => e
```

Comment:
Going to remove the NoMethodError.  The issue was with the behavior with the deleted projects was fixed back in August 2014 29ba57b.  The join with stages actually prevents any deleted projects from causing issues.  The only other potential source of errors would be from the deployer email, which is fixed by using try(:email).